docs/topics/con_configmap-availability.adoc

```diff
@@ -0,0 +1,168 @@
+= Using a ConfigMap to configure your {SpringBoot} application
+//ustory: injecting configuration data into your application using a configmap
+
+:deploymentPlatform: Kubernetes
+
+//openshift docs on configmaps
+//https://docs.openshift.org/latest/dev_guide/configmaps.html
+//https://goo.gl/cQTA2B
+//https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-files
+
+link:http://kubernetes.io/docs/user-guide/configmap/[`ConfigMap`] is a {deploymentPlatform} resource type that allows you to externalize the configuration parameters that you pass to your application in the form of key-value pairs or embedded `application.properties` or `application.yaml` templates.
+The `ConfigMapPropertySource` provided by Spring Cloud Kubernetes searches the deployment configuration of your application for a ConfigMap object in which the `metadata.name` property matches the name of your {SpringBoot} application and applies the values to the deployment configuration of your application.
+The application name is set by the `spring.application.name` property.
+
+Alternatively, you can specify a custom name for your ConfigMap.
+Set the value of the `spring.cloud.kubernetes.config.name` property in the `bootstrap.properties` file of your application project to the name of your ConfigMap.
+This enables the externalized configuration to be recognized and processed by your application.
+
+The examples below demonstrate different ways that configuration data is structured within your ConfigMap YAML template.
+The example below uses the following properties:
+
+* `pool.size.core`
+* `pool.size.maximum`
+
+You can inject configuration settings into your application in different ways depending on how you structure the data within the ConfigMap:
+
+
+// make this a prerequisite:
+// already documented in the Configmap booster doc
+[NOTE]
+====
+Ensure that you have view permissions assigned to the service account of your application.
+Without these permissions, your application will not be able to access the ConfigMap.
+
+[source,bash,options="nowrap",subs="attributes+"]
+--
+oc policy add-role-to-user view system:serviceaccount:$(oc project -q):{app-name} -n $(oc project -q)
+--
+====
+
+* Specify values of individual configuration properties directly in key-value format.
+{deploymentPlatform} processes the ConfigMap and maps the values to your application.
+This structure is suitable for updating specific properties within an otherwise static configuration.
++
+[source,yaml,options="nowrap"]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: demo
+data:
+  pool.size.core: 1
+  pool.size.max: 16
+----
+
+* If your configuration template contains a property named `application.yaml`, {deploymentPlatform} processes the configuration template and creates the ConfigMap.
+On application startup, {deploymentPlatform} checks if the ConfigMap contains a key with one of the following names:
```

Comment:
No it doesn't. Kubernetes doesn't know anything about Spring Boot.

Reply:
Did you mean Spring Cloud Kubernetes @inoxx03?  Also applies to next sentence as well